### PR TITLE
Playback 2023: prompt user to share when a screenshot is taken

### PR DESCRIPTION
**Pull request by leandroalonso**
| 📘 Part of: #1142 |
|:---:|

Detects when a user takes a screenshot and displays an alert.

## To test

1. Make sure you're logged in to an account that has a few episodes listened to this year and last year
2. Go to Profile > Settings > Beta Features > enable `endOfYear`
3. Open stories
4. ✅ Check that the share button is not show for the first story
5. Take a screenshot
6. ✅ You should see an alert and the stories should pause
7. Tap "Not now"
8. ✅ Stories should resume
9. Take another screenshot
10. Tap "Share"
11. ✅ The share sheet should appear
12. Dismiss it
13. ✅ Stories should resume
14. Share the first and the last story by taking a screenshot, tapping Share and saving the image
15. ✅ Check that the stories are saved in Photos

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: podcasts/End of Year/Stories/IntroStory.swift [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import SwiftUI
 
-struct IntroStory: StoryView {
+struct IntroStory: ShareableStory {
```
Comment (leandroalonso) [inline podcasts/End of Year/Stories/IntroStory.swift:3]: I remember that some people share a screenshot of the first and last story so I decided to just handle them. 🤷

File: podcasts/en.lproj/Localizable.strings [hunk 2]
```diff
@@ -3653,6 +3653,12 @@
 /* Subtitle explaining the user why they should susbcribe to Plus in the context of the end of year stories. */
 "eoy_start_your_free_trial" = "Start your Free Trial";
 
+/* Title of an alert displayed to the user asking if they want to share the current story */
+"eoy_share_this_story_title" = "Share this story?";
+
+/* Message of an alert displayed to the user asking if they want to share the current story */
+"eoy_share_this_story_message" = "You can share directly to your favorite app and give a shout out to your favorite shows and creators";
```
Comment (emilylaguna) [inline podcasts/en.lproj/Localizable.strings]: Should we use the wording from [the design](https://www.figma.com/file/lH66LwxxgG8btQ8NrM0ldx/End-of-Year?type=design&node-id=2042-10219&mode=design&t=0JEBt6AJvk8Oj0oi-4)?

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Nice, this works well. Left a comment about the wording.

File: podcasts/en.lproj/Localizable.strings (hunk 2 above)
Comment (leandroalonso) [reply podcasts/en.lproj/Localizable.strings]: Done in ef2edc5796046c26c61586bd2b8bf68209c5f744

**Review by emilylaguna — APPROVED** [review]

**Pull request merged**